### PR TITLE
#656 Setup Structure For Token Validation Middleware

### PR DESCRIPTION
**Pull request by Adeniyii**
This PR is the initial draft for the token validation middleware to protect all application endpoints.

To achieve this, I needed to:
- Assume token was stored on cookie as `authorization` property
- Make verification calls to fake API at `https://test-zuri-core.herokuapp.com/auth/verify/:token`

### CODE REVIEW

File: controllers/visionController.js [hunk 1]
```diff
@@ -1,55 +1,51 @@
...
 const schema = Joi.object({
   title: Joi.string().required(),
-  body: Joi.string().required()
+  body: Joi.string().required(),
 });
 
 // request to get the vision
 exports.getAllVision = async (req, res, next) => {
   const organizationId = '1'; // Would be gotten from zuri main
   const url = `${baseUrl}/data/read/${pluginId}/${collectionName}/${organizationId}`;
```
Comment (Ose-4g) [inline controllers/visionController.js:21]: for Well done. But for the database calls kindly check out the databasehelper.js file. It helps to abstract some of the databse calls.

Comment (Ose-4g) [inline controllers/visionController.js:21]: Please check db/databasehelper.js  Use that to make calls to the database as its a cleaner way to make calls to the db.

**Review by Ose-4g — CHANGES_REQUESTED** [review]

File: controllers/visionController.js (hunk 1 above)
Comment (Adeniyii) [reply controllers/visionController.js:21]: Good morning Ose, I didn't write this controller. This can be changed on a different PR, thanks. @Ose-4g

Comment (Ose-4g) [reply controllers/visionController.js:21]: Alright @Adeniyii . Seeing the changes in those files made me assume you worked on those. I'll merge now.

**Pull request merged**